WordPress/Helpers/WPHookHelper.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WordPressCS\WordPress\Helpers;
+
+use PHPCSUtils\Utils\PassedParameters;
+
+/**
+ * Helper utilities for recognizing functions related to the WP Hook mechanism.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @since   3.0.0 The property in this class was previously contained in the
+ *                `WordPressCS\WordPress\Sniff` class and has been moved here.
+ */
+final class WPHookHelper {
+
+	/**
+	 * A list of functions that invoke WP hooks (filters/actions).
+	 *
+	 * @since 0.10.0
+	 * @since 0.11.0 Changed from public static to protected non-static.
+	 * @since 3.0.0  - Moved from the Sniff class to this class.
+	 *               - The property visibility has changed from `protected` to `private static`.
+	 *                 Use the `get_function_names()` method for access.
+	 *               - The format of the value has changed from a non-relevant boolean to
+	 *                 an array with the parameter position and name(s) for the hook name parameter.
+	 *
+	 * @var array<string, <string, int|string|array>> Function name as key, array with target
+	 *                                                parameter position and name(s) as value.
+	 */
+	private static $hookInvokeFunctions = array(
+		'do_action' => array(
+			'position' => 1,
+			'name'     => 'hook_name',
```

Comment:
In what way would this key-value be used?

Reply:
It is used in the `get_hook_name_param()` function, which is being added in this PR, to allow for supporting PHP 8.0+ function calls using named parameters. The implementation in the two sniffs using this helper class will follow in follow-up PRs.